src/utils/pgwire/src/pg_server.rs

```diff
@@ -200,7 +200,11 @@ pub async fn handle_connection<S, SM>(
         let msg = match pg_proto.read_message().await {
             Ok(msg) => msg,
             Err(e) => {
-                tracing::error!(error = %e.as_report(), "error when reading message");
+                if e.kind() == std::io::ErrorKind::UnexpectedEof {
+                    tracing::warn!("unexpected end of file and it could be a health check");
```

Comment:
Honestly, as a user-facing change, the new message doesn't sound very professional to me. 😢 

I'm considering whether we should remove these `error` logs at all.

This is because the message received here is not directly constructed by the users, but always through clients or drivers. It's our responsibility to get them compatible with RisingWave, and the users can't understand or do anything with these warnings or errors at all. In this sense, it's more like a `debug` level messages to me.

Reply:
+1. I think UnexpectedEof is very precise and "it could be a health check" sounds too casual..

IMO if we want to elegantly handle this case, we may check the number of bytes read for the startup message. If it's 0 bytes, then we can happily return, and fires an error otherwise.

Reply:
It seems we can add a special HealthCheck message to detect it precisely.